src/main/java/org/jabref/JabRefExecutorService.java

```diff
@@ -87,6 +78,29 @@ public boolean executeAndWait(Callable command) {
         }
     }
 
+    /**
+     * This method allows to execute a callable task that provides a return value after the calculation is done.
+     * 
+     * @param command The task to execute.
+     * @return A Future object that provides the returning value.
+     */
+    public <T> Future<T> executeAndReturn(Callable<T> command) {
+        Objects.requireNonNull(command);
+        return executorService.submit(command);
+    }
+    
+    public <T> List<Future<T>> executeAll(Collection<Callable<T>> tasks) {
```

Comment:
According to the discussion at https://stackoverflow.com/a/1250668/873282, all futures should be available. Thus, no need to return something.